### ISSUE TITLE
Non-fullscreen ArFragmernt

### ISSUE THREAD

**Issue by BobbyRaduloff** [google-ar/sceneform-android-sdk#573]
Hello,
I've been trying to create a fragment app with one of the fragments being an ArFragment. I want the status bar and nav bar to be visible but nothing has worked so far. I tried following the advice in #88 , but that didn't work for me. Here's the code I attempted to use:
```
public class NFArFragment extends ArFragment {
    @Override
    protected void onWindowFocusChanged(boolean hasFocus) {
        FragmentActivity activity = getActivity();
        if (hasFocus && activity != null) {
            activity.getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_VISIBLE);
            activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
            activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                                            | WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION
                                            | WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS
                                            | WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
        }
    }
}
```
I did not call super as the poster on #88 suggested and applied the flags he did to no avail. Any help is appreciated.

**Comment by jpeltone** [comment]
Hi, I'm building similar app and struggled with the same. I have BottomNavigationView and one of the fragments shown is ArFragment. If this suits you you can follow the BottomNavigationView example and then in the switch statement do something like this:
...
case R.id.arfragment:
                // Action to perform when Search Menu item is selected.
                arFragment = new ArFragment();
                pushFragment(arFragment, "ARFragment");
...
 /**
         * Method to push any fragment into given id.
         *
         * @param fragment An instance of Fragment to show into the given id.
         */
        protected void pushFragment(Fragment fragment, String tag) {
            if (fragment == null)
                return;

            FragmentManager fragmentManager = getSupportFragmentManager();
            if (fragmentManager != null) {
                FragmentTransaction ft = fragmentManager.beginTransaction();
                if (ft != null) {
                    ft.replace(R.id.rootLayout, fragment, tag);
                    ft.addToBackStack(null);
                    ft.commit();
                    //after transaction you must call the executePendingTransaction
                    fragmentManager.executePendingTransactions();
                }
            }
        }
...

R.id.rootLayout is the layout in your activity.xml you want the fragment to be shown. Hopefully this would something feasible for you too.

**Comment by BobbyRaduloff** [comment]
Thanks for your suggestion @jpeltone, but due to how my codebase is structured, I wasn't able to implement your suggestion. Her's an explanation of how I solved the problem:

First, I was getting the reference to the ArFragment the wrong way. Since the ArFragment resided in a normal Fragment, I needed to call `getChildFragmentManager().findFragmentById(R.id.ux_fragment);`, and not `getSupportFragmentManager().findFragmentById(R.id.ux_fragment);`. After debugging, it turned out that that method was returning `null` all along.

Second, the reference of type `ArFragment` could not be cast to my subclass that I created. To fix this, I went into my `fragment_ar.xml` layout file and replaced the `fragment` tag's name to that of my subclass. This way, `findFragmentById` returns a `NFArFragment` (my subclass) and there are no casting issues.

The actual code for showing the status bar was correct as per #88 , but I was referencing my ArFragment in the wrong way. Hope this helps someone.
